### DISCUSSION TITLE
Internal brotli library call

### DISCUSSION THREAD

**Discussion by dmgnr** [copy/v86#782] [Q&A] [answered]
Yes I recently saw that v86 have a secret Brotli library included for state saving... I forgot where it was 😅
Is there a way to use it myself? to compress image, BIOS, wasm!

**Comment by dmgnr** [comment]
![image](https://user-images.githubusercontent.com/42496449/206397623-0798c9e3-9213-411c-9b58-a4e10c043c53.png)
there found it

**Comment by dmgnr** [comment]
But yes, I still want to use zstd from inside v86...

**Comment by copy** [comment, marked as answer]
See https://github.com/copy/v86/commit/df1a7129f4e1a3877e8803614cf1d24769b87034

**Comment by dmgnr** [comment]
> See https://github.com/copy/v86/commit/df1a7129f4e1a3877e8803614cf1d24769b87034

I see, so you implemented it right into starter module, only decompress??
How do i compress and also get the size?
I cannot install zstd on my system for some reason

**Comment by dmgnr** [comment]
Also my self-contained linux vm is 10mb... But the state is 30mb 🤔🤔
